assets/js/components/simulation/QuestionnaireSimulation.jsx

```diff
@@ -197,18 +197,20 @@ class QuestionnaireSimulation extends Component<Props, State> {
   handleATMessage = (message) => {
     const { projectId, questionnaireId, mode } = this.props
     const { simulation } = this.state
-    if (simulation) {
+    let reply
+
+    if (!simulation) return
+
+    if (message.body) {
```

Comment:
I see from below (`onSendMessage`) that the `message` type can be `Message` or `String`. This still seems a little odd to me. I'm not sure about how it could be improved. Maybe considering a new type of Message? Something like `{ body: "timeout", type: "failure" }`, and then handling them differently based on the `message.type` ?